Mathlib/GroupTheory/SpecificGroups/Cyclic.lean

```diff
@@ -781,6 +781,46 @@ theorem IsCyclic.card_mulAut [Group G] [Finite G] [h : IsCyclic G] :
 
 end ZMod
 
+section powMonoidHom
+
+variable (G)
+
+-- Note. Even though cyclic groups only require `[Group G]`, we need `[CommGroup G]` for
+-- `powMonoidHom` to be defined.
+
+@[to_additive]
+theorem IsCyclic.card_powMonoidHom_range [CommGroup G] [IsCyclic G] [Finite G] (d : ℕ) :
+    Nat.card (powMonoidHom d : G →* G).range = Nat.card G / (Nat.card G).gcd d := by
+  obtain ⟨g, h⟩ := exists_zpow_surjective G
+  have : (powMonoidHom d).range = Subgroup.zpowers (g ^ d) := by
+    rw [show g ^ d = powMonoidHom d g by rfl, ← MonoidHom.map_zpowers,
+      (Subgroup.eq_top_iff' (Subgroup.zpowers g)).mpr h,  ← MonoidHom.range_eq_map]
+  rw [this, Nat.card_zpowers, orderOf_pow, orderOf_eq_card_of_forall_mem_zpowers h]
```

Comment:
Here's a golf. And I wonder if we should have a lemma `orderOf_eq_card_of_zpowers_eq_top`?
```suggestion
  obtain ⟨g, h⟩ := isCyclic_iff_exists_zpowers_eq_top.mp hG
  rw [MonoidHom.range_eq_map, ← h, MonoidHom.map_zpowers, Nat.card_zpowers, powMonoidHom_apply,
    orderOf_pow, orderOf_eq_card_of_forall_mem_zpowers fun g ↦ h.ge (Subgroup.mem_top g)]
```